### PR TITLE
fix(grouping): More classification

### PR DESCRIPTION
**Pull request by untitaker**
Test harness detected overlapping testcases, so I also removed them in a separate commit

* `<compiler-generated>` was annotated on too many frames, so we now go with a list of guaranteed-noise function names
* `Looper.loop` and similar frames are a threadbase. They are used to group ANR which is bad. Instead of categorizing all those random system frames as threadbases, remove systemframes close to threadbase
* libart contains random utility functions so it can't be sentinel frame. Instead pick out a few interesting functions
* `java.lang.Thread`, the entire module, is present in way more frames that aren't threadbases, and as a result the entire stacktrace wasn't grouped by

### CODE REVIEW

**Comment by EvanPurkhiser** [comment]
Would be nice to have a scope in the PR title

**Review by ghost — APPROVED** [review]
looks good

File: src/sentry/grouping/enhancer/enhancement-configs/mobile@2021-04-02.txt [hunk 1]
```diff
@@ -270,7 +275,7 @@ category:malloc +prefix
 category:lock +sentinel +prefix
 category:load +sentinel +prefix
 category:gl +sentinel
-category:gc +sentinel
+category:runtime +sentinel
```
Comment (jan-auer) [inline src/sentry/grouping/enhancer/enhancement-configs/mobile@2021-04-02.txt:278]: We should start adding reasoning in comments to such rules. This one particular, what's the thought behind it? Wouldn't this cause irrelevant runtime frames to be chosen over top app frames in grouping?

File: src/sentry/grouping/enhancer/enhancement-configs/mobile@2021-04-02.txt [hunk 2]
```diff
@@ -88,6 +88,8 @@ family:native function:memcpy category=system
 family:native function:__memcpy category=system
 family:native function:memcmp category=system
 family:native package:CoreFoundation category=system
+family:native package:Foundation category=system
+family:native package:CFNetwork category=system
```
Comment (jan-auer) [inline src/sentry/grouping/enhancer/enhancement-configs/mobile@2021-04-02.txt:92]: Should we just add the full list of [Apple system frameworks](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/OSX_Technology_Overview/SystemFrameworks/SystemFrameworks.html) here?

Comment (untitaker) [reply src/sentry/grouping/enhancer/enhancement-configs/mobile@2021-04-02.txt:92]: I'll create a separate task for this where we get testcases for each of those symbols and potentially have finer-grained categorization. In principle we should alredy classify most of them via their paths on macos, just not on iOS.

File: src/sentry/grouping/enhancer/enhancement-configs/mobile@2021-04-02.txt (hunk 1 above)
Comment (untitaker) [reply src/sentry/grouping/enhancer/enhancement-configs/mobile@2021-04-02.txt:300]: Fair enough, I added a comment

**Review by jan-auer — APPROVED** [review]

**Pull request merged**